extensions/wikia/UserLogin/EmailConfirmationHooks.class.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+class EmailConfirmationHooks {
+
+	/**
+	 * Add JS messages to the output
+	 * @param \OutputPage $out An output object passed from a hook
+	 * @return bool
+	 */
+	public static function onBeforePageDisplay( \OutputPage $out ) {
+		$emailConfirmedParam = F::app()->wg->request->getVal( 'emailConfirmed' );
```

Comment:
`$out->getRequest()->getVal( 'emailConfirmed' );` would be preferred since `OutputPage` is a `ContextSource`. And since you're getting an int, `$out->getRequest()->getInt( 'emailConfirmed' );` and then you can use strict comparisons below (`$emailConfirmedParam === 1`).